hack/build.sh

```diff
@@ -41,55 +41,61 @@ readonly SITE=$PWD/site
 rm -rf site/
 
 if [ "$BUILD_VERSIONS" == "no" ]; then
-  # HEAD to /docs if we're not doing versioning.
-  mkdocs build -f mkdocs.yml -d site/docs
+  # Build to root if we're not doing versioning
+  mkdocs build -f mkdocs.yml -d site
 else
-  # Versioning: pre-release (HEAD): docs => development/
-  cp -r . $TEMP/docs-main
+  # Build latest version to /docs
+  cp -r . "$TEMP/docs-main"
   curl -f -L --show-error https://raw.githubusercontent.com/knative/serving/main/docs/serving-api.md -s > "$TEMP/docs-main/docs/serving/reference/serving-api.md"
   curl -f -L --show-error https://raw.githubusercontent.com/knative/eventing/main/docs/eventing-api.md -s > "$TEMP/docs-main/docs/eventing/reference/eventing-api.md"
-  pushd "$TEMP/docs-main"; mkdocs build -f mkdocs.yml -d $SITE/development; popd
-
-  # Latest release branch to /docs
-  git clone --depth 1 -b ${DOCS_BRANCHES[0]} https://github.com/${GIT_SLUG} "$TEMP/docs-$latest"
-
-  if [ ${latest#*1.} -gt 6 ]; then
-      curl -f -L --show-error https://raw.githubusercontent.com/knative/serving/${DOCS_BRANCHES[0]}/docs/serving-api.md -s > "$TEMP/docs-$latest/docs/serving/reference/serving-api.md"
-      curl -f -L --show-error https://raw.githubusercontent.com/knative/eventing/${DOCS_BRANCHES[0]}/docs/eventing-api.md -s > "$TEMP/docs-$latest/docs/eventing/reference/eventing-api.md"
-  else
-      curl -f -L --show-error https://raw.githubusercontent.com/knative/serving/${DOCS_BRANCHES[0]}/docs/serving-api.md -s > "$TEMP/docs-$latest/docs/reference/api/serving-api.md"
-  curl -f -L --show-error https://raw.githubusercontent.com/knative/eventing/${DOCS_BRANCHES[0]}/docs/eventing-api.md -s > "$TEMP/docs-$latest/docs/reference/api/eventing-api.md"
-  fi
-
-  pushd "$TEMP/docs-$latest"; KNATIVE_VERSION="${VERSIONS[0]}.0" SAMPLES_BRANCH="${DOCS_BRANCHES[0]}" mkdocs build -d $SITE/docs; popd
-
-  # Previous release branches release-$version to /v$version-docs
+  
+  # Create docs directory structure
+  mkdir -p "$SITE/docs"
+  
+  # Build latest docs to /docs
```

Comment:
We were building the next-release docs to `/development/...` and the most-recent-release docs to `/docs/...`.  I think you've changed this to build the next-release docs to `/docs/...`, and not publish docs for the current release anywhere.